File: Lab3/src/updateprofile.py
import tornado.web
import json
import base64

class Handler(tornado.web.RequestHandler):
    def post(self):
        J=json.loads(self.request.body)
        name = J["name"]
        dob = J["dob"]
        email = J["email"]
        image = base64.b64decode(J["image"])
        print("WE GOT:",name,dob,email,image[:20])
        resp={"ok:": True}
        self.write( json.dumps(resp) )

File: Lab3/html/updateprofile.js
"use strict";

function submit(){
    let file = document.getElementById("img").files[0];
    if(!file){
        console.log("No file!");
        return;
    }
    let R = new FileReader();
    R.addEventListener("load", () => {
        let image = btoa(R.result);    //do base64 encoding
        let name = document.getElementById("name").value;
        let dob = document.getElementById("dob").value;
        let email = document.getElementById("email").value;
        let J = {
            name,
            dob,
            email,
            image,
        };
        fetch( "/updateprofile",
            {   method: "POST",
                body: JSON.stringify(J)
            }
        ).then( (resp) => {
            resp.json().then( (J) => {
                console.log("Server said:",J);
            });
        }).catch( (err) => {
            console.log("Uh oh",err);
        })
    });
    R.readAsBinaryString(file);
}